specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,133 @@
+package size is small and the framework carries the dependency.
+
+A class of MSIX packages that are Full Trust or have the packageManagement capability will have
+the permissions to immediately resolve the dependency. These packages are the target for this
+API. These packages will have the capability to deploy packages and satisfy the dependency.
+Since the main packages for the WinAppSDK are expected to be very small, this API has a
+streamlined solution for this scenario, which is to carry the WinAppSDK main packages inside
+the WinAppSDK frameworks, and leverage the full trust capability of the caller to deploy the main
+packages.
+
+
+# Description
+
+When a packaged app using the WinAppSDK is deployed from the Store, it is expected to only have
+the application's main package and frameworks, and the WinAppSDK other main packages may not be
+present, or may not be at the required version. This API enables a developer to detect when the
+WinAppSDK is incomplete and take corrective action.
+
+Since this API is part of the WinAppSDK framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. 
+
+Developers must not assume that the WinAppSDK main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate the WinAppSDK is the correct version and all required packages are present.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of the Windows App SDK I require and are they usable?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Since the main packages are small and the target developer
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+Processes lacking permission to deploy MSIX are out-of-scope at this time.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify all required packages are present
+to a minimum version needed by the loaded framework.  A process should call GetStatus() after
+process initialization and before using Windows App Runtime content (e.g. in main() or WinMain()).
+
+Developers call this method without parameters, as all information about the version, channel, and
+architecture are derived from the current WARfwk loaded. This method is good to identify if there
+is possible work required in the form of package install which may take several seconds to
```

Comment:
As a follow-on, what threads are these APIs expected to be called on?  The samples should probably include a little more context about whether these are called in WinMain or in a message loop or on a background thread.  Maybe show a more complete example using "await/co_await" to specifically show it?

Reply:
It sort of depends on the app. At the simplest implementation, a single Initialize() call could be on WinMain, as in the sample. The first time the app launches, it _may_ need to deploy packages, in which case it may take a few seconds. Subsequent launches should not have that issue. 

The times where Initialize() actually deploys packages should be extremely limited to the first time the app is deployed to the system. Even other users should have a faster registration time than the initial first user deployment. This performance issue may not be a significant concern to some developers, in which case this solution is probably sufficient.

If you want a more robust handling of the potential performance issue, instead call GetStatus(), check the result, and if the result is not good, call Initialize() on another thread and you can handle the time delay more elegantly.